src/viam/sdk/components/camera/camera.cpp

```diff
@@ -24,6 +25,68 @@ API API::traits<Camera>::api() {
     return {kRDK, kComponent, "camera"};
 }
 
+std::vector<unsigned char> Camera::encode_depth_map(const Camera::depth_map& m) {
+    if (m.depth_values.size() != m.width * m.height) {
```

Comment:
Does this stay here, or move into the `Camera::depth_map` constructor? Not sure if assertions like this belong in structs

Reply:
That's a good question! My instinct is to keep it here, I think it makes documenting the error cases more consistent with what's happening in the code base currently. But I'm actually not sure what's more idiomatic for C++ broadly speaking, @acmorrow likely knows more!

Reply:
Ahh another reason to do it here is, someone could theoretically modify the `depth_map` after constructing it, thus calling here with an invalid depth_map even if they constructed a valid one. So yeah, let's leave it as is!